### PR TITLE
Migration to Spine 0.8

### PR DESCRIPTION
**Pull request by IlliaShepilov**
Summary of changes:

- Dependencies between `BoundedContext`, repositories and enrichers are unleashed.
- `BoundedContext` definition moved to the separate class.
- Tests are reworked according to the `Junit 5` features.
- `TaskAggregate` is separated into parts.
- Created `TaskAggregateRoot`.
- Created separate module for the test util data.
- Renamed `TaskLabelId` to `LabelId`.

### CODE REVIEW

**Comment by IlliaShepilov** [comment]
@armiol PTAL.

**Comment by armiol** [comment]
@IlliaShepilov please fix the typos in the description.

File: api-java/src/main/java/org/spine3/examples/todolist/EnrichmentNotFoundException.java [hunk 1]
```diff
@@ -27,6 +27,8 @@
  */
 public class EnrichmentNotFoundException extends RuntimeException {
 
+    private static final long serialVersionUID = 6015609539020998331L;
```
Comment (armiol) [inline api-java/src/main/java/org/spine3/examples/todolist/EnrichmentNotFoundException.java]: Why `1L`  is not good in this case?

File: api-java/src/main/java/org/spine3/examples/todolist/c/aggregates/AggregateHelper.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ */
+
+package org.spine3.examples.todolist.c.aggregates;
+
+import org.spine3.examples.todolist.TaskStatus;
+
+/**
+ * @author Illia Shepilov
+ */
+class AggregateHelper {
```
Comment (armiol) [inline api-java/src/main/java/org/spine3/examples/todolist/c/aggregates/AggregateHelper.java]: JavaDocs please.

Also, the naming is very wide for this content, i.e. just a single utility method to ease the exception propagation. Consider renaming.

File: api-java/src/main/java/org/spine3/examples/todolist/c/aggregates/AggregateHelper.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    private AggregateHelper() {
+    }
+
+    static String generateExceptionMessage(TaskStatus currentStatus, TaskStatus newStatus) {
```
Comment (armiol) [inline api-java/src/main/java/org/spine3/examples/todolist/c/aggregates/AggregateHelper.java]: Again, the name does not read properly.

File: api-java/src/main/java/org/spine3/examples/todolist/c/aggregates/FailureHelper.java [hunk 4]
```diff
@@ -58,6 +58,7 @@
  *
  * @author Illia Shepilov
  */
+@SuppressWarnings("OverlyCoupledClass")
```
Comment (armiol) [inline api-java/src/main/java/org/spine3/examples/todolist/c/aggregates/FailureHelper.java]: Why is it OK to suppress this warning? Don't you think it's better to split the class somehow?

File: api-java/src/main/java/org/spine3/examples/todolist/c/aggregates/FailureHelper.java [hunk 5]
```diff
@@ -382,8 +383,7 @@ static void throwCannotUpdateLabelDetailsFailure(TaskLabelId labelId, ValueMisma
      * according to the passed parameters.
      *
      * @param taskId the ID of the task
-     * @throws CannotUpdateTaskWithInappropriateDescription
-     * the {@code CannotUpdateTaskWithInappropriateDescription} failure
+     * @throws CannotUpdateTaskWithInappropriateDescription the {@code CannotUpdateTaskWithInappropriateDescription} failure
```
Comment (armiol) [inline api-java/src/main/java/org/spine3/examples/todolist/c/aggregates/FailureHelper.java]: Why duplicate the failure name? 

The comment applies to this and the similar cases.

File: api-java/src/main/java/org/spine3/examples/todolist/c/aggregates/TaskAggregateRoot.java [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+    @SuppressWarnings("StaticVariableMayNotBeInitialized")
+    private static BoundedContext boundedContext;
+
+    static {
+        initBoundedContextIfNeeded();
+    }
+
+    /**
+     * Obtains and sets the {@link BoundedContext} singleton instance,
+     * if the {@link Environment} is not test environment.
```
Comment (armiol) [inline api-java/src/main/java/org/spine3/examples/todolist/c/aggregates/TaskAggregateRoot.java]: `is a non-test environment`.

File: api-java/src/main/java/org/spine3/examples/todolist/c/aggregates/TaskAggregateRoot.java [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+     * Creates a new {@link TaskAggregateRoot} instance.
+     *
+     * @param id the ID of the aggregate
+     */
+    protected TaskAggregateRoot(TaskId id) {
+        super(boundedContext, id);
+    }
+
+    /**
+     * Returns {@link TaskAggregateRoot} instance according to the {@code TaskId}.
```
Comment (armiol) [inline api-java/src/main/java/org/spine3/examples/todolist/c/aggregates/TaskAggregateRoot.java]: The JavaDocs say the method `Returns`, but instead it `Creates`. Something is not right.

File: api-java/src/main/java/org/spine3/examples/todolist/c/aggregates/TaskDefinitionPart.java [hunk 8]
```diff
@@ -97,34 +90,32 @@
...
  * @author Illia Shepilov
  */
-// The methods annotated with {@link Assign} are declared {@code private} by design.
-// Also, such methods must take a single {@code Event} as a parameter.
-// However, it is unused in some cases due to the business rules.
-// E.g. {@code TaskRemoved} event is explicit enough by itself,
-// so it is not needed to use any of its field values. So the parameter left unused.
-@SuppressWarnings("unused")
-public class TaskAggregate extends Aggregate<TaskId, Task, Task.Builder> {
+@SuppressWarnings({"OverlyCoupledClass", "ClassWithTooManyMethods"})
```
Comment (armiol) [inline api-java/src/main/java/org/spine3/examples/todolist/c/aggregates/TaskDefinitionPart.java]: Consider refactoring this class into a number of classes to make the warnings disappear without an explicit suppression.

File: api-java/src/main/java/org/spine3/examples/todolist/c/aggregates/TaskDefinitionPart.java [hunk 9]
```diff
@@ -211,7 +202,7 @@ public TaskAggregate(TaskId id) {
         final TaskPriority actualPriority = state.getPriority();
         final TaskPriority expectedPriority = priorityChange.getPreviousValue();
 
-        boolean isEquals = actualPriority.equals(expectedPriority);
+        boolean isEquals = actualPriority == expectedPriority;
```
Comment (armiol) [inline api-java/src/main/java/org/spine3/examples/todolist/c/aggregates/TaskDefinitionPart.java]: Why is it not resolved? Either a Codacy rule should be tweaked or the issue resolved.

File: api-java/src/main/java/org/spine3/examples/todolist/c/aggregates/TaskLabelsPart.java [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.examples.todolist.c.aggregates;
```
Comment (armiol) [inline api-java/src/main/java/org/spine3/examples/todolist/c/aggregates/TaskLabelsPart.java]: `aggregates` — should it be plural or singular? Let's discuss.

File: api-java/src/main/java/org/spine3/examples/todolist/c/aggregates/TaskLabelsPart.java [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+import static org.spine3.examples.todolist.c.aggregates.FailureHelper.throwCannotRemoveLabelFromTaskFailure;
+import static org.spine3.examples.todolist.c.aggregates.TaskFlowValidator.isValidAssignLabelToTaskCommand;
+import static org.spine3.examples.todolist.c.aggregates.TaskFlowValidator.isValidRemoveLabelFromTaskCommand;
+
+/**
+ * The aggregate managing the state of a {@link TaskLabels}.
+ *
+ * @author Illia Shepilov
+ */
+@SuppressWarnings("unused") // The methods annotated with {@link Assign} are declared {@code private} by design.
```
Comment (armiol) [inline api-java/src/main/java/org/spine3/examples/todolist/c/aggregates/TaskLabelsPart.java]: Please think before you copy-paste. They are not `private` in this class.

File: api-java/src/main/java/org/spine3/examples/todolist/context/EventBusEnricher.java [hunk 12]
```diff
@@ -0,0 +1,186 @@
...
+import org.spine3.examples.todolist.repositories.TaskDefinitionRepository;
+import org.spine3.examples.todolist.repositories.TaskLabelsRepository;
+import org.spine3.server.event.EventBus;
+
+/**
+ * Performs all necessary initializations to create and add enricher fields for the {@link EventBus}.
+ *
+ * @author Illia Shepilov
+ */
+@SuppressWarnings("Guava") // as we use Guava Function until we migrate the whole framework to Java 8.
```
Comment (armiol) [inline api-java/src/main/java/org/spine3/examples/todolist/context/EventBusEnricher.java]: Please try to avoid `we`.

File: api-java/src/main/java/org/spine3/examples/todolist/context/EventBusEnricher.java [hunk 13]
```diff
@@ -0,0 +1,186 @@
...
+public class EventBusEnricher {
+
+    private static final String REPOSITORY_IS_NOT_INITIALIZED = "Repository is not initialized.";
```
Comment (armiol) [inline api-java/src/main/java/org/spine3/examples/todolist/context/EventBusEnricher.java]: Allow an empty line after the constant declaration block.

File: api-java/src/main/java/org/spine3/examples/todolist/context/EventBusEnricher.java [hunk 14]
```diff
@@ -0,0 +1,186 @@
...
+    private final TodoListRepositoryProvider repositoryProvider;
```
Comment (armiol) [inline api-java/src/main/java/org/spine3/examples/todolist/context/EventBusEnricher.java]: Why do you still need it? Let's discuss.

File: api-java/src/main/java/org/spine3/examples/todolist/context/EventBusEnricher.java [hunk 15]
```diff
@@ -0,0 +1,186 @@
...
+     * Creates a new builder for (@code EventBusEnricher).
+     *
+     * @return new builder instance
+     */
+    public static Builder newBuilder() {
+        return new Builder();
+    }
+
+    /**
+     * A builder for producing {@code EventEnricherSupplier} instances.
```
Comment (armiol) [inline api-java/src/main/java/org/spine3/examples/todolist/context/EventBusEnricher.java]: Remove `producing`.

File: api-java/src/main/java/org/spine3/examples/todolist/context/TodoListBoundedContext.java [hunk 16]
```diff
@@ -0,0 +1,122 @@
...
+
+/**
+ * Serves for creation the {@link BoundedContext} instances.
+ *
+ * @author Illia Shepilov
+ */
+public class TodoListBoundedContext {
+
+    /** The name of the Bounded Context. */
+    private static final String NAME = "Todo";
```
Comment (armiol) [inline api-java/src/main/java/org/spine3/examples/todolist/context/TodoListBoundedContext.java]: `Todo`? really? Please think of a proper name.

File: api-java/src/main/java/org/spine3/examples/todolist/context/TodoListRepositoryProvider.java [hunk 17]
```diff
@@ -0,0 +1,80 @@
...
+
+import javax.annotation.Nullable;
+import java.util.Optional;
+
+/**
+ * Serves as a repository provider for all the application components.
+ *
+ * @author Illia Shepilov
+ */
+class TodoListRepositoryProvider {
```
Comment (armiol) [inline api-java/src/main/java/org/spine3/examples/todolist/context/TodoListRepositoryProvider.java]: Looks stale. I think it can be removed at all.

File: api-java/src/main/java/org/spine3/examples/todolist/c/aggregates/TaskDefinitionPart.java (hunk 9 above)
Comment (IlliaShepilov) [inline api-java/src/main/java/org/spine3/examples/todolist/c/aggregates/TaskDefinitionPart.java]: Inspections profile is configured in opposite way. Which one is correct?

**Comment by IlliaShepilov** [comment]
@armiol PTAL.

File: model/src/main/proto/todolist/model.proto [hunk 18]
```diff
@@ -62,21 +61,31 @@ message Task {
     // Optional. Has no default value.
     google.protobuf.Timestamp due_date = 5;
 
-    // Task labels.
+    // Task status.
     //
-    // Task can have any number of labels assigned, i.e. [0; +inf).
-    // Optional. Empty by default.
-    repeated TaskLabelId label_ids = 6;
+    // Optional. Defaults to the `TaskStatus.TS_UNDEFINED`.
```
Comment (armiol) [inline model/src/main/proto/todolist/model.proto]: Is it optional in the real life? Let's discuss.

File: testutil/src/main/java/org/spine3/examples/todolist/testdata/TestTaskLabelCommandFactory.java [hunk 19]
```diff
@@ -52,8 +52,18 @@ private TestTaskLabelCommandFactory() {
      * @return {@link CreateBasicLabel} instance
      */
     public static CreateBasicLabel createLabelInstance() {
+        final CreateBasicLabel result = createLabelInstance(LABEL_ID);
+        return result;
+    }
+
+    /**
+     * Provides default {@link CreateBasicLabel} event instance.
```
Comment (armiol) [inline testutil/src/main/java/org/spine3/examples/todolist/testdata/TestTaskLabelCommandFactory.java]: It does not.

File: api-java/src/main/java/org/spine3/examples/todolist/c/aggregate/FailureHelper.java [hunk 20]
```diff
@@ -0,0 +1,437 @@
...
+import org.spine3.examples.todolist.c.failures.CannotUpdateTaskWithInappropriateDescription;
+
+/**
+ * Utility class for working with failures.
+ *
+ * @author Illia Shepilov
+ */
+class FailureHelper {
+
+    static final String TASK_DELETED_OR_COMPLETED_EXCEPTION_MESSAGE = "Command cannot be applied " +
```
Comment (IlliaShepilov) [inline api-java/src/main/java/org/spine3/examples/todolist/c/aggregate/FailureHelper.java]: I am using that constant in the `TaskDefinitionPart` to avoid literal duplications.

File: api-java/src/main/java/org/spine3/examples/todolist/c/aggregate/FailureHelper.java [hunk 21]
```diff
@@ -0,0 +1,437 @@
...
+    }
+
+    /**
+     * Constructs and throws the {@link CannotReopenTask} failure according to the passed parameters.
+     *
+     * @param taskId  the ID of the task
+     * @param message the exception message
+     * @throws CannotReopenTask the failure to throw
+     */
+    static void throwCannotReopenTaskFailure(TaskId taskId, String message) throws CannotReopenTask {
```
Comment (IlliaShepilov) [inline api-java/src/main/java/org/spine3/examples/todolist/c/aggregate/FailureHelper.java]: I discussed it with you when I was defining the failures.

File: api-java/src/main/java/org/spine3/examples/todolist/c/aggregate/FailureHelper.java [hunk 22]
```diff
@@ -0,0 +1,437 @@
...
+        /**
+         * Constructs and throws the {@link CannotCreateTaskWithInappropriateDescription} failure
+         * according to the passed parameters.
+         *
+         * @param taskId the ID of the task
+         * @throws CannotCreateTaskWithInappropriateDescription the failure to throw
+         */
+        static void throwCannotCreateTaskWithInappropriateDescriptionFailure(TaskId taskId)
+                throws CannotCreateTaskWithInappropriateDescription {
+            final UnsuccessfulTaskCommand commandFailed =
```
Comment (IlliaShepilov) [inline api-java/src/main/java/org/spine3/examples/todolist/c/aggregate/FailureHelper.java]: It is not failure, it is dto, which wrapped by failure.

File: api-java/src/test/java/org/spine3/examples/todolist/q/projection/DraftTasksViewProjectionTest.java [hunk 23]
```diff
@@ -0,0 +1,382 @@
...
+        final TaskDraftCreated taskDraftCreatedEvent = taskDraftCreatedInstance();
+        projection.on(taskDraftCreatedEvent);
+    }
+
+    @Nested
+    @DisplayName("TaskDraftCreated event")
+    class TaskDraftCreatedEvent {
+
+        @Test
+        @DisplayName(ADDS_TASK_VIEW_TO_STATE)
```
Comment (IlliaShepilov) [inline api-java/src/test/java/org/spine3/examples/todolist/q/projection/DraftTasksViewProjectionTest.java]: To avoid literal duplications.

**Comment by IlliaShepilov** [comment]
@armiol PTAL.

File: api-java/src/main/java/org/spine3/examples/todolist/context/TodoListBoundedContext.java [hunk 24]
```diff
@@ -0,0 +1,107 @@
...
+        final BoundedContext boundedContext = createBoundedContext();
+
+        final TaskDefinitionRepository taskDefinitionRepo = new TaskDefinitionRepository(boundedContext);
+        final LabelAggregateRepository labelAggregateRepo = new LabelAggregateRepository(boundedContext);
+        final TaskLabelsRepository taskLabelsRepo = new TaskLabelsRepository(boundedContext);
+
+        boundedContext.register(taskDefinitionRepo);
+        boundedContext.register(taskLabelsRepo);
+        boundedContext.register(labelAggregateRepo);
+        boundedContext.register(new MyListViewRepository(boundedContext));
```
Comment (armiol) [inline api-java/src/main/java/org/spine3/examples/todolist/context/TodoListBoundedContext.java]: Please have the registration unified. Atm, three of the repos are created above and then registered, the other three are created in-place.

File: api-java/src/main/java/org/spine3/examples/todolist/context/TodoListEventEnricher.java [hunk 25]
```diff
@@ -0,0 +1,185 @@
...
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Serves as class which adds enrichment fields to the {@link EventBus}.
+ *
+ * @author Illia Shepilov
+ */
+@SuppressWarnings("Guava") // because com.google.common.base.Function is used
+                           // until the migration of Spine to Java 8 is performed.
+public class TodoListEventEnricher {
```
Comment (armiol) [inline api-java/src/main/java/org/spine3/examples/todolist/context/TodoListEventEnricher.java]: That isn't an `Enricher` as we discussed previously. Please rename/refactor.
I'd suggest `TodoListEnrichmentConfiguration`. And rename `Builder.build()` to `Builder.apply(EventBus bus)`.

File: api-java/src/test/java/org/spine3/examples/todolist/c/aggregate/LabelAggregateTest.java [hunk 26]
```diff
@@ -0,0 +1,217 @@
...
+
+    private static LabelId createLabelId() {
+        final LabelId result = LabelId.newBuilder()
+                                      .setValue(newUuid())
+                                      .build();
+        return result;
+    }
+
+    @Nested
+    @DisplayName("CreateBasicLabel command should be interpret by LabelAggregate and")
```
Comment (armiol) [inline api-java/src/test/java/org/spine3/examples/todolist/c/aggregate/LabelAggregateTest.java]: See the past-participle-related section [here](https://en.wiktionary.org/wiki/interpret#Verb).

File: api-java/src/test/java/org/spine3/examples/todolist/c/aggregate/TaskLabelsPartTest.java [hunk 27]
```diff
@@ -0,0 +1,233 @@
...
+            assertEquals(LabelAssignedToTask.class, messageList.get(0)
+                                                               .getClass());
+            final LabelAssignedToTask labelAssignedToTask = (LabelAssignedToTask) messageList.get(0);
+
+            assertEquals(taskId, labelAssignedToTask.getTaskId());
+            assertEquals(labelId, labelAssignedToTask.getLabelId());
+        }
+
+        @Test
+        @DisplayName("throw CannotAssignLabelToTask failure when try to assign the label to the deleted task")
```
Comment (armiol) [inline api-java/src/test/java/org/spine3/examples/todolist/c/aggregate/TaskLabelsPartTest.java]: `when trying`.

File: testutil/src/main/java/org/spine3/examples/todolist/testdata/TestTaskEventFactory.java [hunk 28]
```diff
@@ -0,0 +1,302 @@
...
+     */
+    public static LabelledTaskRestored labelledTaskRestoredInstance() {
+        final LabelledTaskRestored result = LabelledTaskRestored.newBuilder()
+                                                                .setTaskId(TASK_ID)
+                                                                .setLabelId(LABEL_ID)
+                                                                .build();
+        return result;
+    }
+
+    public static class UpdateEventFactory {
```
Comment (armiol) [inline testutil/src/main/java/org/spine3/examples/todolist/testdata/TestTaskEventFactory.java]: `UpdateEvents`. No need to duplicate `...Factory`.

Comment (armiol) [inline testutil/src/main/java/org/spine3/examples/todolist/testdata/TestTaskEventFactory.java]: Also, why only `Update` events grouped? Please think of some structure for the others as well — in case you still want to group them anyway.

File: testutil/src/main/java/org/spine3/examples/todolist/testdata/TestTaskLabelsCommandFactory.java [hunk 29]
```diff
@@ -0,0 +1,86 @@
...
+import org.spine3.examples.todolist.TaskId;
+import org.spine3.examples.todolist.c.commands.AssignLabelToTask;
+import org.spine3.examples.todolist.c.commands.RemoveLabelFromTask;
+
+/**
+ * A factory of the task label commands for the test needs.
+ *
+ * @author Illia Shepilov
+ */
+public class TestTaskLabelsCommandFactory {
```
Comment (armiol) [inline testutil/src/main/java/org/spine3/examples/todolist/testdata/TestTaskLabelsCommandFactory.java]: Please take a look at the most recent `testutil` module. There are some utilities to make the command creation easier (as well as aggregate testing etc.) Maybe we could move to the latest Spine right away and use those instead of manually created testing routines.

**Comment by armiol** [comment]
@IlliaShepilov `Spine` is a title. Please fix in the PR.

**Comment by armiol** [comment]
@IlliaShepilov also, as to the PR description. If you are using a bulleted list and delimit the items with a semicolon, then the next item should not start with a capital letter.

E.g. (wrong)

* First;
* Second;
* Third.

(correct)
* first;
* second;
* third.

(or — correct)
* First.
* Second.
* Third.

File: testutil/src/main/java/org/spine3/examples/todolist/testdata/TestTaskLabelsCommandFactory.java (hunk 29 above)
Comment (IlliaShepilov) [inline testutil/src/main/java/org/spine3/examples/todolist/testdata/TestTaskLabelsCommandFactory.java]: I have looked at it. I need to create custom factory methods to build some commands anyway, but some of them can be replaced and simplified. 
So I propose to do it during the separate PR in time of migration to the latest Spine.

**Comment by IlliaShepilov** [comment]
@armiol PTAL.

File: model/src/main/proto/todolist/c/failures.proto [hunk 30]
```diff
@@ -29,115 +29,116 @@ option java_generate_equals_and_hash = true;
 
 import "spine/annotations.proto";
 
+import "todolist/identifiers.proto";
 import "todolist/values.proto";
 
-// Fire the failure when cannot update task description.
+// Fire the failure when cannot update the task description.
```
Comment (armiol) [inline model/src/main/proto/todolist/c/failures.proto]: Let me provide you with a sample:

```
The failure to update a task description.

It is thrown if an attempt is made to update the task in either DRAFT or COMPLETED state.
```

BTW, will it be thrown if the task is deleted? Please mention that if so.

File: client/src/test/java/org/spine3/examples/todolist/client/CreateBasicTaskTest.java [hunk 31]
```diff
@@ -0,0 +1,100 @@
...
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.spine3.examples.todolist.testdata.TestTaskCommandFactory.DESCRIPTION;
+
+/**
+ * @author Illia Shepilov
+ */
+@DisplayName("After execution CreateBasicTask command")
```
Comment (armiol) [inline client/src/test/java/org/spine3/examples/todolist/client/CreateBasicTaskTest.java]: `After execution of CreateBasicTask command` (`of` is missing).
Please fix where applicable.

File: api-java/src/test/java/org/spine3/examples/todolist/q/projection/ProjectionTest.java [hunk 32]
```diff
@@ -0,0 +1,38 @@
...
+package org.spine3.examples.todolist.q.projection;
+
+import org.spine3.examples.todolist.TaskListId;
+
+import static org.spine3.base.Identifiers.newUuid;
+
+/**
+ * @author Illia Shepilov
+ */
+abstract class ProjectionTest {
```
Comment (armiol) [inline api-java/src/test/java/org/spine3/examples/todolist/q/projection/ProjectionTest.java]: Please have a JavaDoc description.

File: api-java/src/test/java/org/spine3/examples/todolist/q/projection/ProjectionHelperTest.java [hunk 33]
```diff
@@ -32,35 +33,38 @@
...
-    public void have_private_constructor() {
-        assertTrue(hasPrivateUtilityConstructor(ProjectionHelper.class));
+    @DisplayName("have the private constructor")
+    public void havePrivateConstructor() {
+        assertTrue(hasPrivateParameterlessCtor(ProjectionHelper.class));
     }
 
     @Test
-    public void remove_task_view_from_state_by_label_id() {
+    @DisplayName("remove TaskView form state by label ID")
```
Comment (armiol) [inline api-java/src/test/java/org/spine3/examples/todolist/q/projection/ProjectionHelperTest.java]: Please review ALL your `form` typos. It should be `from`.

File: api-java/src/test/java/org/spine3/examples/todolist/c/aggregate/definition/UpdateTaskPriorityTest.java [hunk 34]
```diff
@@ -0,0 +1,185 @@
...
+        } catch (Throwable e) {
+            @SuppressWarnings("ThrowableResultOfMethodCallIgnored") // Need it for checking.
+            final Throwable cause = Throwables.getRootCause(e);
+            assertTrue(cause instanceof CannotUpdateTaskPriority);
+        }
+    }
+
+    @Test
+    @DisplayName("throw CannotUpdateTaskPriority failure " +
+            "when it is trying to update the priority for the completed task")
```
Comment (armiol) [inline api-java/src/test/java/org/spine3/examples/todolist/c/aggregate/definition/UpdateTaskPriorityTest.java]: `upon an attempt to update the priority of the completed task`
Please review the similar cases.

File: model/src/main/proto/todolist/c/failures.proto (hunk 30 above)
Comment (IlliaShepilov) [inline model/src/main/proto/todolist/c/failures.proto]: It should be ...`either DELETED or COMPLETED`.
Sorry, my inattention.

**Comment by IlliaShepilov** [comment]
@armiol PTAL.

File: model/src/main/proto/todolist/c/failures.proto [hunk 35]
```diff
@@ -32,112 +32,112 @@ import "spine/annotations.proto";
 import "todolist/identifiers.proto";
 import "todolist/values.proto";
 
-// Fire the failure when cannot update the task description.
-// Failure fires when task state either DRAFT or COMPLETED.
+// The failure to update a task description.
+// It is thrown if an attempt is made to update the task in either DELETED or COMPLETED state.
```
Comment (armiol) [inline model/src/main/proto/todolist/c/failures.proto]: I have inserted an empty line in my sample for a reason.

File: model/src/main/proto/todolist/c/failures.proto [hunk 36]
```diff
@@ -32,112 +32,112 @@ import "spine/annotations.proto";
...
 message CannotUpdateTaskDueDate {
 
     // Failed the due date update.
     TaskDueDateUpdateFailed update_failed = 1;
 }
 
-// Fire the failure when cannot update the label details.
-// Failure fires when the expected label details does not match the actual.
+// The failure to update a label details.
+// It is thrown if an attempt is made to update the label when the expected label details does not match the actual.
```
Comment (armiol) [inline model/src/main/proto/todolist/c/failures.proto]: `do not match`.

File: model/src/main/proto/todolist/c/failures.proto [hunk 37]
```diff
@@ -32,112 +32,112 @@ import "spine/annotations.proto";
...
 message CannotUpdateLabelDetails {
 
     // Failed the label details update.
     LabelDetailsUpdateFailed update_failed = 1;
 }
 
-// Fire the failure when cannot assign label to task.
-// Failure fires when task state either DRAFT or COMPLETED.
+// The failure to assign label to task.
+// It is thrown if an attempt is made to assign the label to task in either DELETED or COMPLETED state.
```
Comment (armiol) [inline model/src/main/proto/todolist/c/failures.proto]: `the task`.

File: model/src/main/proto/todolist/c/failures.proto [hunk 38]
```diff
@@ -32,112 +32,112 @@ import "spine/annotations.proto";
...
 message CannotAssignLabelToTask {
 
     // Failed the label assigning to the task.
     AssignLabelToTaskFailed assign_label_failed = 1;
 }
 
-// Fire the failure when cannot remove label from task.
-// Failure fires when task state either DRAFT or COMPLETED.
+// The failure to remove label from task.
+// It is thrown if an attempt is made to remove the label from task in either DELETED or COMPLETED state.
```
Comment (armiol) [inline model/src/main/proto/todolist/c/failures.proto]: `the task`

File: model/src/main/proto/todolist/c/failures.proto [hunk 39]
```diff
@@ -32,112 +32,112 @@ import "spine/annotations.proto";
...
 message CannotFinalizeDraft {
 
     // Failed the draft finalization.
     FinalizeDraftFailed finalize_draft_failed = 1;
 }
 
-// Fire the failure when cannot update task with inappropriate description.
-// Failure fires when the task description contains less than three symbols.
+// The failure to update a task description.
+// It is thrown if an attempt is made to update the task description when description has less than three symbols.
```
Comment (armiol) [inline model/src/main/proto/todolist/c/failures.proto]: `the description`

File: model/src/main/proto/todolist/c/failures.proto [hunk 40]
```diff
@@ -32,112 +32,112 @@ import "spine/annotations.proto";
...
 message CannotUpdateTaskWithInappropriateDescription {
 
     // Failed the description update.
     DescriptionUpdateFailed update_failed = 1;
 }
 
-// Fire the failure when cannot create task with inapproprriate descriprion.
-// Failure fires when the task description contains less the three symbols.
+// The failure to create a task.
+// It is thrown if an attempt is made to create the task when task description has less than three symbols.
```
Comment (armiol) [inline model/src/main/proto/todolist/c/failures.proto]: `when the suggested description`

**Comment by IlliaShepilov** [comment]
@armiol PTAL.
